measurements/pages/__init__.py

```diff
@@ -215,32 +215,31 @@ def decompress_autoclaved(
         intra_off,
         report_size
     ):
-    url = urljoin(current_app.config['AUTOCLAVED_BASE_URL'], autoclaved_filename)
     def generator():
         try:
+            url = urljoin(current_app.config['AUTOCLAVED_BASE_URL'], autoclaved_filename)
             headers = {"Range": "bytes={}-{}".format(frame_off, frame_off + total_frame_size)}
             r = requests.get(url, headers=headers, stream=True)
-            streamed_data = 0
-            while streamed_data < report_size:
-                for chunk in lz4framed.Decompressor(r.raw):
-                    # I am at the beginning of the stream and I need to send some
-                    # offset into the data
-                    if streamed_data == 0 and intra_off > 0:
-                        chunk = chunk[intra_off:]
-                    # Avoid streaming more data than I should
-                    if (streamed_data + len(chunk)) > report_size:
-                        chunk = chunk[:abs(streamed_data - report_size)]
-                    yield chunk
+            beginning = True
+            # Create a copy because we are in a closure
+            to_read = report_size
```

Comment:
I thought that floats() are immutable... I don't understand the reason to make a copy :-(

Reply:
That's how closures work in python. As soon as you assign the enclosed variable (as we do below in `to_read -= len(d)`) it's converted to a local reference and since that conversion is happening within the scope of the closure the `while` loop will given an unboundvariable error.